dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,511 @@
+            "build_with_datadog_dep.gradle.kts",
+            appBuildGradleFile
+        )
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", ":samples:app:assembleRelease")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":samples:app:assembleRelease")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assembleDebug`() {
+        // Given
+        stubGradleBuildFromResourceFile(
+            "build_with_datadog_dep.gradle.kts",
+            appBuildGradleFile
+        )
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(":samples:app:assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":samples:app:assembleDebug")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assembleDebug { project with library module }`() {
+        // Given
+        stubGradleBuildFromResourceFile(
+            "build_with_lib_module_attached.gradle.kts",
+            appBuildGradleFile
+        )
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(":samples:app:assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":samples:app:assembleDebug")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assembleDebug { build cache }`() {
+        // Given
+        stubGradleBuildFromResourceFile(
+            "build_with_datadog_dep.gradle.kts",
+            appBuildGradleFile
+        )
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", ":samples:app:assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":samples:app:assembleDebug")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    // The assembleRelease fails for configuration cache and not because of our
+    // code but because of these tasks: collect[Flavour1][Flavour2]ReleaseDependencies. For this
+    // reason the below 2 scenarios are only tested for the `debug` build type.
+
+    @Test
+    fun `M fail W assembleDebug { configuration cache, checkProjectDependecies enabled }`() {
+
+        // This test should fail because our `DdCheckSdkDepsTask` is not following the
+        // --configuration-cache best practices (we are using the project inside) and so far we
+        // could not easily solve that.
```

Comment:
Wait, you mean that for now the `assembleDebug` fails but because there's a bug in our project, and not because it's expected to fail? Then we shouldn't have this test at all. We need to write prescriptive tests, not descriptive ones.
Put in other word: 
- if we want the `assembleDebug` task to fail in this specific scenario (using the `build_with_datadog_dep.gradle.kts` file and running the task with `--configuration-cache`), then yes we can keep this test;
- if we want the task to succeed, but it fails for technical reasons (untestable, non compliance on our part with the configuration cache or anything else), then either we make it succeed in the test or we do write the test for success, mark it as `@Ignored` and create a task in JIRA to make it pass.